presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/ApiScope.java

```diff
@@ -0,0 +1,28 @@
+package com.teradata.prestomanager.controller;
+
+import com.teradata.prestomanager.controller.api.ControllerConfigAPI;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public enum ApiScope
+{
+
+    CLUSTER, WORKERS, COORDINATOR, INVALID;
+
+    private static final Logger LOGGER = LogManager.getLogger(ControllerConfigAPI.class);
+
+    public static ApiScope fromString(String s)
+    {
+        if (s == null) {
```

Comment:
Because of what seems to be a bug in Jersey, this will never be called if `ApiScope` is used as a parameter.

But this should return `null` or a default scope, not an invalid scope. (And if it really has to be invalid, I would recommend just adding `NullPointerException` to the `catch`-block a few lines down.)